plugins/action/common/prepare_plugins/prep_108_route_control.py

```diff
@@ -0,0 +1,318 @@
+# Copyright (c) 2024 Cisco Systems, Inc. and its affiliates
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# SPDX-License-Identifier: MIT
+
+import re
+from jinja2 import ChainableUndefined, Environment, FileSystemLoader
+from ansible_collections.ansible.utils.plugins.filter import ipaddr, hwaddr
+from ....plugin_utils.helper_functions import hostname_to_ip_mapping, data_model_key_check
+
+
+class PreparePlugin:
+    """
+    Class PreparePlugin
+    """
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+        self.keys = []
+
+    def prepare(self):
+        """
+        function to prepare data for route_control
+        """
+        templates_path = self.kwargs['templates_path']
+        model_data = self.kwargs['results']['model_extended']
+        default_values = self.kwargs['default_values']
+
+        template_filename = "ndfc_route_control.j2"
+
+        env = Environment(
+            loader=FileSystemLoader(templates_path),
+            undefined=ChainableUndefined,
+            lstrip_blocks=True,
+            trim_blocks=True,
+        )
+
+        env.filters["ipaddr"] = ipaddr.ipaddr
+        env.filters["hwaddr"] = hwaddr.hwaddr
+        template = env.get_template(template_filename)
+
+        parent_keys = ['vxlan', 'overlay_extensions', 'route_control', 'route_maps']
+        dm_check = data_model_key_check(model_data, parent_keys)
+
+        # Check Route-Maps
```

Comment:
Ok this is easier to read/understand now so thanks for refactoring.  Would it be safe to say that `self.route_maps` and `self.ipv[4|6]_access_list` are actually updating and transforming model data?  If so can se update the comments to indicate we are actually updating and transforming the data and add `update_` to the function names to be more clear when calling the function.

Reply:
rename functions for route-maps and IP ACLs.
* update_route_maps()
* update_ip_access_lists()